Content.Shared/Mobs/Components/MobStateComponent.cs

```diff
@@ -14,92 +27,176 @@ namespace Content.Shared.Mobs.Components;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool CanEmote(MobState? State = null) => GetOverride(State).Emoting ?? GetParams(State)?.Emoting ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool CanThrow(MobState? State = null) => GetOverride(State).Throwing ?? GetParams(State)?.Throwing ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool CanPickUp(MobState? State = null) => GetOverride(State).PickingUp ?? GetParams(State)?.PickingUp ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool CanPull(MobState? State = null) => GetOverride(State).Pulling ?? GetParams(State)?.Pulling ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool CanAttack(MobState? State = null) => GetOverride(State).Attacking ?? GetParams(State)?.Attacking ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool CanUse(MobState? State = null) => GetOverride(State).Using ?? GetParams(State)?.Using ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool CanPoint(MobState? State = null) =>GetOverride(State).Pointing ?? GetParams(State)?.Pointing ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool IsConscious(MobState? State = null) => GetOverride(State).IsConscious ?? GetParams(State)?.IsConscious ?? false;
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool IsDowned(MobState? State = null) => GetOverride(State).ForceDown ?? GetParams(State)?.ForceDown ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool ShouldDropItems(MobState? State = null) => GetOverride(State).DropItemsOnEntering ?? GetParams(State)?.DropItemsOnEntering ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool IsIncapacitated(MobState? State = null) => GetOverride(State).Incapacitated ?? GetParams(State)?.Incapacitated ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool IsThreatening(MobState? State = null) => GetOverride(State).Threatening ?? GetParams(State)?.Threatening ?? true; // :)
+    
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool CanEquipSelf(MobState? State = null) => GetOverride(State).CanEquipSelf ?? GetParams(State)?.CanEquipSelf ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool CanUnequipSelf(MobState? State = null) => GetOverride(State).CanUnequipSelf ?? GetParams(State)?.CanUnequipSelf ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool CanEquipOther(MobState? State = null) => GetOverride(State).CanEquipOther ?? GetParams(State)?.CanEquipOther ?? false;
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool CanUnequipOther(MobState? State = null) => GetOverride(State).CanUnequipOther ?? GetParams(State)?.CanUnequipOther ?? false;
```

Comment:
Holy fuck this is amazing. I'm taking notes.

Reply:
:flushed: